hw6/hw6/ViewController.swift

```diff
@@ -0,0 +1,149 @@
+        }
+        
+        print(evenArray)
+        print(unevenArray)
+        
+        // 2) Дан массив опциональных строк. Посчитайте, сколько букв каждая буква встречается во всех словах.
+        let optionalStringArray: [String?] = ["aaabbbcde", "nil", "hhhhhh"]
+        
+        countNumberRepeatedLettersOptionalStringArray(optionalStringArray)
+        
+        // 3) Реализуйте математические функции (+, -, *, /, возведение в степень) для опциональных чисел (тип данных на ваш выбор). Используйте if let, guard let и ?? каждую хотя бы 1 раз.
+        print(add(3, to: 5))
+        print(add(4, to: nil))
+        print(multiply(3, 5))
+        print(multiply(3, nil))
+        print(myPow(3, 5))
+        print(myPow(3, nil))
+        print(subtraction(3, 5))
+        print(subtraction(3, nil))
+        print(division(3, 5))
+        print(division(3, nil))
+        
+        // 4) * Создайте функцию, которая проверяет является ли переданная строка палиндромом. Строка может быть как одним словом (mom) так и предложением (Too bad I hid a boot). В предложении вы должны игнорировать пробелы, большие или маленькие это буквы и знаки препинания при определении полиндрома.
+        
+        print(polyndrom("Too bad I hid a boot"))
+   }
+
+    func add(_ firstNumber: Int?, to secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber + secondNumber)
+    }
+    
+    func multiply(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        if let firstNumber = firstNumber,
+           let secondNumber = secondNumber {
+            return String(firstNumber * secondNumber)
+        }
+        return "Одно из значений не число"
+    }
+    
+    func myPow(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(pow(Double(firstNumber), Double(secondNumber)))
+    }
+    
+    func subtraction(_ firstNumber: Int?, _ secondNumber: Int?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber - secondNumber)
+    }
+    
+    func division(_ firstNumber: Double?, _ secondNumber: Double?) -> String {
+        guard let firstNumber = firstNumber,
+              let secondNumber = secondNumber else {
+            return "Одно из значений не число"
+        }
+        return String(firstNumber / secondNumber)
+    }
+    
+    func createCharacterArray(_ stringArray: [String?]) -> [Character] {
+        var characterArray = [Character]()
+
+        for element in stringArray {
+            guard let element = element else {
+                return ["E"]
```

Comment:
Why have you decided to go with this letter in this case? Maybe it's better just to return an empty array?